### PR TITLE
Add Calculated Fields class for Attendee model

### PR DESCRIPTION
**Pull request by tn3rb**
## Problem this Pull Request solves
need to be able to retrieve attendee related data via the API that is not immediately available via the models. For example: Attendee Gravatars

## How has this been tested
only in usage in the Event Attendees Gutenberg block branch

### CODE REVIEW

File: core/libraries/rest_api/calculations/Attendee.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     */
+    public static function userAvatar(array $wpdb_row, WP_REST_Request $request, Base $controller)
+    {
+        $attendee = Attendee::getAttendeeObject($wpdb_row);
+        if (! $attendee instanceof EE_Attendee) {
+            return '';
+        }
+        $avatar = get_avatar_url($attendee->email());
```
Comment (mnelson4) [inline core/libraries/rest_api/calculations/Attendee.php]: if we're already needing to dig into the array to get the `Attendee_CPT.ID` entry, we could save a little processing by instead grabbing the `Attendee_Meta.ATT_email` value, and passing that to `get_avatar_url()`, no?

File: core/libraries/rest_api/CalculatedModelFields.php [hunk 2]
```diff
@@ -82,6 +83,9 @@ protected function generateNewMapping()
                 'Registration' => array(
                     'datetime_checkin_stati' => $registration_class,
                 ),
+                'Attendee' => array(
+                    'userAvatar' => $attendee_class,
```
Comment (mnelson4) [inline core/libraries/rest_api/CalculatedModelFields.php]: this string `userAvatar` is what REST API clients will put in the querystring. The convention so far has been to have those be snake case. However, the method on the class can remain camelcase- `CalcualtedModelFields::retrieveCalculatedFieldValue()` takes care of translating the snake-case string into a camelcase one to determine which method to use.

**Comment by tn3rb** [comment]
> the private method was unneeded

oh I left that in on purpose because I figured it would be handy if not necessary for any further additions to this class. No problem removing it though. thnx

**Comment by tn3rb** [comment]
oh and went ahead and merged this since it does not interact with any other code

**Pull request merged**